packages/cozy-client/src/helpers.spec.js

```diff
@@ -141,3 +149,128 @@ describe('generateWebLink', () => {
     ).toEqual(`https://drive.alice.cozy.tools/public/#/files/432432`)
   })
 })
+
+describe('rootCozyUrl', () => {
+  beforeAll(() => {
+    enableFetchMocks()
+
+    // Valid URLs, not necessarily valid root Cozy URLs though
+    fetch.mockIf(
+      'https://camillenimbus.mycozy.cloud/.well-known/change-password'
+    )
+    fetch.mockIf(
+      'https://camillenimbus-drive.mycozy.cloud/.well-known/change-password',
+      { status: 401 }
+    )
+    fetch.mockIf('https://camillenimbus.com/.well-known/change-password', {
+      status: 200
+    })
+    fetch.mockIf('https://camille-nimbus.com/.well-known/change-password')
+    fetch.mockIf('https://camillenimbus.com:666/.well-known/change-password')
+    fetch.mockIf('http://camille-nimbus.com/.well-known/change-password')
+    fetch.mockIf('https://camille.nimbus.com/.well-known/change-password')
+    fetch.mockIf(
+      'https://camille-drive.nimbus.com/.well-known/change-password',
+      { status: 401 }
+    )
+    fetch.mockIf('http://cozy.localhost:8080/.well-known/change-password')
+  })
+
+  afterAll(() => {
+    disableFetchMocks()
+  })
+
+  it('should handle cozy-hosted https', async () => {
+    await expect(
+      rootCozyUrl(new URL('https://camillenimbus.mycozy.cloud'))
+    ).resolves.toEqual(new URL('https://camillenimbus.mycozy.cloud'))
```

Comment:
let's add a const variable = new URL('https://camillenimbus.mycozy.cloud')

```suggestion
    ).resolves.toEqual(expectedRootCozyUrl)
```

Reply:
The value will be different in a number of tests so I don't see a major improvement in settings this constant.